src/graphics/TextureFragment.cpp

```diff
@@ -0,0 +1,43 @@
+#include "TextureFragment.h"
+#include "utils.h"
+#include "../engine/constants.h"
+
+TextureFragment::TextureFragment(TextureKeeper texture, Rectangle source, float scale) {
+  _texture = texture;
+  _getTextureCoordFromImageCoord(source, scale);
+}
+
+TextureFragment::TextureFragment(const char* filename, Rectangle source, float scale) {
+  _texture = TextureKeeper(filename);
+  _getTextureCoordFromImageCoord(source, scale);
+}
+
+void TextureFragment::_getTextureCoordFromImageCoord(Rectangle source, float scale) {
```

Comment:
Feels like you should `assert` division by zero here

Reply:
Good catch. I need to check if texture dimentions are not 0. If it's not - then texture is not loaded yet, or loaded incorrectly